http4k-multipart/src/main/kotlin/org/http4k/multipart/DiskLocation.kt

```diff
@@ -21,9 +21,7 @@ interface DiskLocation : Closeable {
                         File.createTempFile(
                             filename ?: (UUID.randomUUID().toString() + "-"),
                             ".tmp", diskDir
-                        ).apply {
-                            deleteOnExit()
```

Comment:
I think `deleteOnExit` should also be removed on line 17.

Reply:
I think that one is ok because it applies to the temp directory as a whole (for which there is only one per JVM process)

Reply:
`deleteOnExit` is being called on `Files.createTempDirectory("http4k-mp")`, which creates a new temp directory in `java.io.tempdir` for every invocation, each with a unique suffix.